### PR TITLE
Adding equals and hashcode methods to PanacheEntity

### PR DESCRIPTION
**Pull request by rhuan080**
Signed-off-by: rhuan080 <rhuan080@gmail.com>

Adding equals and hashcode methods to PanacheEntity

### CODE REVIEW

**Comment by gastaldi** [comment]
Can you please squash your commits? Thanks!

File: extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntity.java [hunk 1]
```diff
@@ -39,5 +41,25 @@ public String toString() {
         return this.getClass().getSimpleName() + "<" + id + ">";
     }
 
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (Objects.isNull(object) || getClass() != object.getClass()) {
```
Comment (nimo23) [inline extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntity.java]: use `object == null`.

Comment (rhuan080) [reply extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntity.java]: I'm so sorry, but why? Objects.isNull(object) executes it:
public static boolean isNull(Object obj) { return obj == null; }

Objects.isNull(object) is more semantic.

Comment (nimo23) [reply extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/PanacheEntity.java]: https://github.com/quarkusio/quarkus/pull/8241#discussion_r399723086

**Comment by FroMage** [comment]
Note that this was already rejected by @loicmathieu in #7046

**Comment by Sanne** [comment]
why multiple PRs for the same change?

Please avoid that, and best to check for existing issues first as well.

**Pull request closed without merging**